src/workerd/api/global-scope.c++

```diff
@@ -230,6 +231,8 @@ kj::Promise<DeferredProxy<void>> ServiceWorkerGlobalScope::request(kj::HttpMetho
   bool useDefaultHandling;
   KJ_IF_SOME(h, exportedHandler) {
     KJ_IF_SOME(f, h.fetch) {
+      jsg::AsyncContextFrame::StorageScope envStorageScope(
```

Comment:
Hmm, you've instrumented every event entrypoint in global-scope, but I think this doesn't cover custom events, including RPC?

Instrumenting every entrypoint feels error-prone. Can we make the initial assignment just apply to the _whole worker_, until such a point as `withEnv()` is used to set a different value?

Reply:
That's a bit difficult given where we initialize the bindingScope in worker.c++. I did try to minimize it but wasn't able to get it working. Before this is moved from draft, however, I plan to get that part figured out so we can minimize the instrumentation.

Reply:
Yep, considered that. Once we're sure of the overall behavior we want I'll go back and clean this bit up.

Reply:
Btw, given just a jsg::Lock and no IoContext, what is the idiomatic way of getting to the `Worker::Impl::env`?

Reply:
Uhhhhhhh.... good question. I guess I'm not sure there is one. Might need to add something to JSG.

Reply:
Ok, I've got something in mind but it'll take a bit to work up. Will do that tomorrow.

Reply:
Ok, updated the PR to store the env on jsg::Lock rather than in Worker::Impl and it works just fine. Was able to remove all of the instrumentation points and have the lookup fallback as needed.